Yoast/wordpress-seo
# Fix ads being displayed for Premium users

## Summary

This PR can be summarized in the following changelog entry:

* Fixes a bug where Premium users would be shown ads when following a certain path through the SEO menu.

## Relevant technical choices:

* Due to the complexity of the code, the amount of tests that could be added, is minimal. I'll be making an issue for us to look into solving some of the complexity.

## Test instructions

This PR can be tested by following these steps:

* Please first follow the steps described in the issue.
* Test this branch and ensure that the user keeps seeing ads in their environment.
* Merge this branch into Premium and activate Premium.
* Follow the steps again that were described in the issue and note that the ads no longer show up.

## Quality ensurance


Fixes https://github.com/Yoast/wordpress-seo-premium/issues/1550
